### PR TITLE
layers: Add remaining VK_EXT_nested_command_buffer

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/6702

### CODE REVIEW

**Review by pdaniell-nv — APPROVED** [review]
LGTM. Many thanks!

File: layers/core_checks/cc_cmd_buffer.cpp [hunk 1]
```diff
@@ -915,6 +925,16 @@ bool CoreChecks::PreCallValidateCmdExecuteCommands(VkCommandBuffer commandBuffer
                     }
                 }
 
+                if ((sub_cb_state.beginInfo.flags & VK_COMMAND_BUFFER_USAGE_SIMULTANEOUS_USE_BIT) &&
+                    !enabled_features.nestedCommandBufferSimultaneousUse) {
+                    skip |=
+                        LogError("VUID-vkCmdExecuteCommands-nestedCommandBufferSimultaneousUse-09378", pCommandBuffers[i], cb_loc,
+                                 "(%s) is was recorded with VkCommandBufferBeginInfo::flag including "
```
Comment (artem-lunarg) [inline layers/core_checks/cc_cmd_buffer.cpp:932]: "is was recorded" typo

File: layers/core_checks/cc_cmd_buffer.cpp [hunk 2]
```diff
@@ -1209,6 +1229,19 @@ bool CoreChecks::PreCallValidateCmdExecuteCommands(VkCommandBuffer commandBuffer
                         }
                     }
                 }
+
+                // spec: "A maxCommandBufferNestingLevel of UINT32_MAX means there is no limit to the nesting level"
+                if (enabled_features.nestedCommandBuffer &&
+                    phys_dev_ext_props.nested_command_buffer_props.maxCommandBufferNestingLevel != UINT32_MAX) {
+                    if ((sub_cb_state.nesting_level + 1) >
```
Comment (artem-lunarg) [inline layers/core_checks/cc_cmd_buffer.cpp:1236]: If someone creates a command buffer with nested level of UINT32_MAX, `sub_cb_state.nesting_level + 1` will result in 0, which won't trigger an error. This is not something one can do in practice, but this mostly to simplify code validation (you still need to think a little bit about this).
` if (sub_cb_state.nesting_level >=  phys_dev_ext_props.nested_command_buffer_props.maxCommandBufferNestingLevel)` does not wrap around.

Comment (spencer-lunarg) [reply layers/core_checks/cc_cmd_buffer.cpp:1236]: good catch

**Review by artem-lunarg — APPROVED** [review]

**Pull request merged**